pkg/sloop/processing/watch.go

```diff
@@ -124,3 +126,21 @@ func toWatchTableKeyPrefix(ts *timestamp.Timestamp, kind string, namespace strin
 
 	return typed.NewWatchTableKey(untyped.GetPartitionId(timestamp), kind, namespace, name, time.Time{}), nil
 }
+
+func GetUidForWatchEntry(tables typed.Tables, txn badgerwrap.Txn, kind string, namespace string, name string, timestamp time.Time) (string, error) {
+	watchKeyComparator := typed.NewWatchTableKeyComparator(kind, namespace, name, timestamp)
+	seekKey := queries.GetSeekKey(watchKeyComparator, time.Now())
```

Comment:
Do we need a nil check here?

Reply:
All the fields have already been checked in parent function